ytsync/sources/youtubeVideo.go

```diff
@@ -99,7 +97,12 @@ func (v YoutubeVideo) getAbbrevDescription() string {
 func (v YoutubeVideo) download() error {
 	videoPath := v.getFilename()
 
-	_, err := os.Stat(videoPath)
+	err := os.Mkdir(v.dir+"/"+v.id, 0750)
+	if err != nil && !strings.Contains(err.Error(), "file exists") {
```

Comment:
Actually, this is not meant to check if a directory exists, instead it's meant to report any error other than "file exists". This is due to the fact that each video has its own subdirectory to avoid name conflicts and if for some reasons the video was previously downloaded (ie on a retry attempt) the directory doesn't have to be re-created.

If however you think I should check for the directory before calling mkdir then I will adapt the code.